### PR TITLE
files_sharing: Add OpenAPI spec

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

From https://github.com/nextcloud/server/pull/36666. Adds the necessary annotations and descriptions.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -1434,7 +1444,7 @@ public function pendingShares(): DataResponse {
 
 			try {
 				$formattedShare = $this->formatShare($share, $node);
-				$formattedShare['status'] = $share->getStatus();
+				$formattedShare['status'] = $share->getStatus(); // TODO: This seems to be a bug because the formatShare method writes the user status into this field
```
Comment (provokateurin) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:1447]: TODO: Discuss with @juliushaertl since he introduced this

Comment (provokateurin) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1447]: @skjnldsv @come-nc What do you think about this?

Comment (juliushaertl) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1447]: Yes, you're right, though when I added it formatShare didn't handle this:

- Added in Jan 2020 https://github.com/nextcloud/server/commit/445274cf93dbae31c6995008dcc0f9834c05d9ca
- formatShare added in Aug 2020 https://github.com/nextcloud/server/pull/22109

Comment (provokateurin) [reply apps/files_sharing/lib/Controller/ShareAPIController.php]: Ok, thanks!

**Review by skjnldsv — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Pull request merged**